subsys/net/ip/net_context.c

```diff
@@ -2426,6 +2434,22 @@ int net_context_recv(struct net_context *context,
 	return 0;
 }
 
+int net_context_update_recv_wnd(struct net_context *context,
+				int delta)
+{
+#if defined(CONFIG_NET_TCP)
+	if (!context->tcp) {
+		NET_ERR("context->tcp == NULL");
+		return -EPROTOTYPE;
+	}
+
+	context->tcp->recv_wnd += delta;
```

Comment:
> Well, we can, but it will make code much less pretty ;-). Do we really need to do that now? ;-)

I think we need to, otherwise this will be forgotten :)

The recv_wnd is uint16_t recv_wnd; (actually it should be u16_t, missed this on my first review), and we just cannot +- some user specified int value.